### ISSUE TITLE
UnusedPrivateClass has false positive behavior for deserialized items

### ISSUE THREAD

**Issue by imanushin** [arturbosch/detekt#2158]
Code:
```
   fun readId() = 
       jacksonObjectMapper().readValue<Response>(content).id

  private class Response(
      val id: Int
  )
```

## Expected Behavior
```Response``` class is used, so rule "UnusedPrivateClass" skips it.

## Observed Behavior
Rule "UnusedPrivateClass" warns, that ```Response``` class is private and is not used.

## Context
Private classes for deserialization are very useful to hide interops inside deserialization wrappers.

## Your Environment
* Version of detekt used: 1.2.1
* Version of Gradle used (if applicable): 6.0.1
* Operating System and version: Windows 7 x64
* Link to your project (if it's a public repository): NDA

**Comment by schalkms** [comment]
Thanks for reporting this bug.
